sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -906,30 +906,27 @@ struct _pi_kernel : _pi_object {
 
   // Hash function object for the unordered_set below.
   struct Hash {
-    size_t operator()(
-        const std::unordered_map<void *, MemAllocRecord>::iterator &It) const {
-      return std::hash<void *>()(It->first);
+    size_t operator()(const std::pair<void *const, MemAllocRecord> *P) const {
+      return std::hash<void *>()(P->first);
     }
   };
 
   // If kernel has indirect access we need to make a snapshot of all existing
   // memory allocations to defer deletion of these memory allocations to the
   // moment when kernel execution has finished.
-  // We store iterators because iterator is not invalidated by insert/delete for
-  // std::map.
-  // Why need to take a snapshot instead of just reference-counting the
-  // allocations, because picture of active allocations can change during kernel
-  // execution (new allocations can be added) and we need to know which memory
-  // allocations were retained by this kernel to release them (and don't touch
-  // new allocations) at kernel completion.
-  // Same kernel may be submitted several times and retained allocations may be
-  // different at each submission. That's why we have a set of memory
-  // allocations here and increase ref count only once even if kernel is
-  // submitted many times. We don't want to know how many times and which
-  // allocations were retained by each submission. We release all allocations
-  // in the set only when SubmissionsCount == 0.
-  std::unordered_set<std::unordered_map<void *, MemAllocRecord>::iterator, Hash>
-      MemAllocs;
+  // We store pointers to the elements because pointers are not invalidated by
+  // insert/delete for std::unordered_map (iterators are invalidated). We need
```

Comment:
Could you point me to the C++ standard about this please?

Reply:
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf
23.2.5/9 page 765

Reply:
Thanks!